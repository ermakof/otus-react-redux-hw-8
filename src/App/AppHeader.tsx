import React, { FC, memo } from 'react';
import Panel from '@src/layout/Panel';
import AppRouter from '@src/App/AppRouter';

const AppHeader: FC = () => (
  <Panel role="topPanel">
    <AppRouter />
  </Panel>
);

export default memo(AppHeader);
