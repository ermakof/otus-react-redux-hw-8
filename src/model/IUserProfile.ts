export interface IUserProfile {
  login: string;
  password: string;
  token?: string;
}
