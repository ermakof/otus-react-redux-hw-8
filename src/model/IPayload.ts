import { IUserProfile } from '@src/model/IUserProfile';

interface IPayload {
  selectedCell?: number;
  cellId?: number;
  gameFieldSize?: number;
  gameLevel?: string;
  userProfile?: IUserProfile;
}

export default IPayload;
