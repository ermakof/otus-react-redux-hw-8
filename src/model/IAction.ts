import IPayload from './IPayload';
import { IState } from '@src/model/IState';

export interface IAction {
  type: string;
  payload?: IPayload | IState;
}
