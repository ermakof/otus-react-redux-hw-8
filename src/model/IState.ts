import ICellInfo from '@src/model/ICellInfo';
import { IUserProfile } from '@src/model/IUserProfile';

export interface IState {
  gameFieldSize: number;
  gameFieldPercentFilled: number;
  gameFieldData: Array<ICellInfo>;
  gameLevel: string;
  selectedCell?: number;
  userProfile?: IUserProfile;
  isLoading?: boolean;
}
