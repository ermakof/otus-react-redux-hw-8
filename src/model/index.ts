import { IState } from '@src/model/IState';
import { IAction } from '@src/model/IAction';
import { IUserProfile } from '@src/model/IUserProfile';

export { IAction, IState, IUserProfile };
