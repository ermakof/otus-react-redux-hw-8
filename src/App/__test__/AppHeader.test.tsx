import React from 'react';
import { render, screen } from '@testing-library/react';
import AppHeader from '@src/App/AppHeader';
import Store from '@src/store';
import { IAction } from '@src/model';

let dispatch: (action: IAction) => void;

describe('AppHeader', () => {
  beforeEach(() => {
    dispatch = jest.fn();
  });

  it('Render authorised <AppHeader>', () => {
    const state = {
      gameLevel: '1',
      gameFieldSize: 3,
      gameFieldPercentFilled: 10,
      gameFieldData: [1, 0, 0, 0, 0, 0, 0, 0, 0],
      userProfile: { login: '123', password: '123', token: '7a0cbf93-aa13-4b50-8a41-97ddbfba00d5' },
    };
    const { asFragment } = render(
      <Store.Provider value={{ dispatch, state }}>
        <AppHeader />
      </Store.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    const title = screen.getByText(/Lines/gi);
    expect(title).toBeInTheDocument();
    const button = screen.getByRole(/^buttonLogout$/gi);
    expect(button).toBeInTheDocument();
    const select = screen.getByRole(/select/gi);
    expect(select).toBeInTheDocument();
  });

  it('Render unauthorised <AppHeader>', () => {
    const state = {
      gameLevel: '1',
      gameFieldSize: 3,
      gameFieldPercentFilled: 10,
      gameFieldData: [1, 0, 0, 0, 0, 0, 0, 0, 0],
    };
    const { asFragment } = render(
      <Store.Provider value={{ dispatch, state }}>
        <AppHeader />
      </Store.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    const title = screen.getByText(/Lines/gi);
    expect(title).toBeInTheDocument();
    const button = screen.getByRole(/buttonLogin/gi);
    expect(button).toBeInTheDocument();
  });
});
