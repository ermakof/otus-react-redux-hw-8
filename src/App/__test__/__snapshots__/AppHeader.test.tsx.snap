// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader Render authorised <AppHeader> 1`] = `
<DocumentFragment>
  <div
    class="css-lgrtma"
    role="topPanel"
  >
    <div
      class="css-ov1ktg"
    >
      <p
        class="css-gbrjra"
      >
        Lines 123
      </p>
      <div
        class="css-lgrtma"
        role="userForm"
      >
        <button
          class="css-tv5zn1"
          role="buttonReset"
        >
          Reset
        </button>
        <select
          class="css-tv5zn1"
          name="select"
          role="select"
        >
          <option
            role="option"
            value="1"
          >
            Падаван
          </option>
          <option
            role="option"
            value="2"
          >
            Джедай
          </option>
          <option
            role="option"
            value="3"
          >
            Мастер-джедай
          </option>
          <option
            role="option"
            value="4"
          >
            Член Совета Силы
          </option>
        </select>
        <button
          class="css-tv5zn1"
          role="buttonLogout"
        >
          Выйти
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppHeader Render unauthorised <AppHeader> 1`] = `
<DocumentFragment>
  <div
    class="css-lgrtma"
    role="topPanel"
  >
    <div
      class="css-ov1ktg"
    >
      <p
        class="css-gbrjra"
      >
        Lines
      </p>
      <div
        class="css-115ni0s"
        role="authForm"
      >
        <form
          action="#"
        >
          <label
            for="login"
          >
            Имя
          </label>
          <input
            id="login"
            name="login"
            placeholder="Введите логин"
            role="login"
            value=""
          />
          <label
            for="password"
          >
            Пароль
          </label>
          <input
            id="password"
            name="password"
            placeholder="Введите пароль"
            role="password"
            value=""
          />
          <button
            disabled=""
            role="buttonLogin"
            type="submit"
          >
            Войти
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
